### PR TITLE
Release tracking PR - v0.30.0

### PR DESCRIPTION
**Pull request by tcharding**
Add changelog notes and bump the version number to v0.30.0.

## TODO - pre-merge

- [x] Release `bitcoin_hashes` 0.12: https://github.com/rust-bitcoin/rust-bitcoin/pull/1694
- [x] Release secp 0.27: https://github.com/rust-bitcoin/rust-secp256k1/pull/588
  - https://github.com/rust-bitcoin/rust-secp256k1/pull/590
- [x] Update `secp256k1` dependency to use newly released v0.27: https://github.com/rust-bitcoin/rust-bitcoin/pull/1714 
- [x] Merge
  - ~https://github.com/rust-bitcoin/rust-bitcoin/pull/1696~
  - https://github.com/rust-bitcoin/rust-bitcoin/pull/1695
  -  https://github.com/rust-bitcoin/rust-bitcoin/pull/1111
- [x] If time permits merge these:
  - #1710 
  - #1705 
  - #1713 
- [x] Set the release date in changelog header
- [x] And merge these:
  - https://github.com/rust-bitcoin/rust-bitcoin/pull/1721
  - https://github.com/rust-bitcoin/rust-bitcoin/pull/1720
  - https://github.com/rust-bitcoin/rust-bitcoin/pull/1719
  - https://github.com/rust-bitcoin/rust-bitcoin/pull/1717  

## TODO  - post release
- [ ] Release the blogpost: https://github.com/rust-bitcoin/www.rust-bitcoin.org/pull/2
   - ~Set the date in the blog post to match the date 0.30 is released~

### CODE REVIEW

**Comment by tcharding** [comment]
Force pushed some more changelog notes re updating dependencies we control.

**Comment by Kixunil** [comment]
Note for Optech: should be published after this is merged and the crate is out. Should be soon.

**Comment by apoelstra** [comment]
@tcharding would you be willing to open a PR to do the actual update?

**Comment by apoelstra** [comment]
Oh, never mind, this *is* the PR -- except we just need a PR to update the rust-secp dep. I can do that.

**Comment by apoelstra** [comment]
Done: https://github.com/rust-bitcoin/rust-bitcoin/pull/1714

**Comment by tcharding** [comment]
Are we happy with the taproot API improvements under the "API improvements" header and not in their own section? (Mentioning because we just added a separate taproot section to the blog post.)

**Comment by tcharding** [comment]
@apoelstra I've set the date to Sunday the 19th March so if we release in the next day or two the "set date" checkbox in PR description can be considered done.

File: bitcoin/CHANGELOG.md
Comment (Kixunil) [inline bitcoin/CHANGELOG.md:14]: Unsized `Script` is not mentioned here but I think it should since it's quite large and disruptive

**Comment by Kixunil** [comment]
> Are we happy with the taproot API improvements under the "API improvements" header and not in their own section? (Mentioning because we just added a separate taproot section to the blog post.)

If you mean in changelog then yes, they don't have to be the same.

File: bitcoin/CHANGELOG.md
Comment (tcharding) [reply bitcoin/CHANGELOG.md:3]: I did this vague link on purpose to introduce folks to the new website.

File: bitcoin/CHANGELOG.md
Comment (tcharding) [reply bitcoin/CHANGELOG.md:14]: Added here and also in the "New types" section.

**Comment by tcharding** [comment]
Force push is just the mentions of `ScriptBuf`.

File: bitcoin/CHANGELOG.md
Comment (Kixunil) [reply bitcoin/CHANGELOG.md:3]: Having a separate "Hey, we have a website!" would make more sense then.

Comment (tcharding) [reply bitcoin/CHANGELOG.md:3]: Sure thing, will add.

**Comment by tcharding** [comment]
Rebased on master and added initial link to the new website. Used direct link to blogpost. We have a chicken and egg problem with the blog post and changelog now, we (I) want to add links to online docs to the blog post which implies code released (changelog notes released) which link to the blogpost.

**Comment by Kixunil** [comment]
IMO start by releasing blog first because less people will accidentally stumble upon it.

**Comment by tcharding** [comment]
Aight, I rekon we are good to pull the trigger then - either @apoelstra to do it (blogpost and crate release) or I can be online to do it with you Andrew at normal time. I'm getting pretty excited about this release FTR.

**Comment by tcharding** [comment]
Bother, too excited it would seem. We still need to merge #1714 and then rebase this one.

**Comment by tcharding** [comment]
Updated the release date in changelog to 21st

**Review by apoelstra — DISMISSED** [review]
ACK 99d816948adf03633ae3aac1be64fed45628416c

**Comment by tcharding** [comment]
Needs an ack from you @Kixunil!

**Comment by tcharding** [comment]
Perhaps give @sanket1729 a chance to ack as well?

**Comment by apoelstra** [comment]
If we update this (which we should, unless we get two more ACKs on this commit quickly), we may also want to mention https://github.com/rust-bitcoin/rust-bitcoin/pull/1715 in the changelog (which I just merged)

**Comment by junderw** [comment]
I just noticed something missing from all the float work. See #1720

It's a small change, so I would hope it can get in. Sorry for the last minute. I was reviewing the format PR and noticed Header had a passthrough method.

**Comment by Kixunil** [comment]
Such a shame, 21st March is my lucky day, would've been be really cool to release today. :D

I'm OK with fixing it up but since #1715 merged I'd also like to make `Denomination` `non_exhaustive`.

**Review by Kixunil — DISMISSED** [review]
ACK 31f0f832ad5f13852cf96f3f4fb94d9256822866

@tcharding had a good point about timezones, it's worth releasing with imperfect changelog.

**Review by sanket1729 — APPROVED** [review]
ACK 31f0f832ad5f13852cf96f3f4fb94d9256822866

File: bitcoin/CHANGELOG.md
Comment (tcharding) [reply bitcoin/CHANGELOG.md:60]: I moved the `FromHexStr` line up to "API improvements" section and change performance section to:

- Performance improvements:
  - [Remove needless allocation from BIP-158 encoding](https://github.com/rust-bitcoin/rust-bitcoin/pull/1146)
  - [Implement fast hex encoding](https://github.com/rust-bitcoin/rust-bitcoin/pull/1268) (usage added in a [later PR](https://github.com/rust-bitcoin/rust-bitcoin/pull/1476))

File: bitcoin/CHANGELOG.md
Comment (tcharding) [reply bitcoin/CHANGELOG.md:43]: Changed to 

  - Renamed `Script` to `ScriptBuf` and added a new unsized type [`Script`](https://github.com/rust-bitcoin/rust-bitcoin/pull/1155)

File: bitcoin/CHANGELOG.md
Comment (tcharding) [reply bitcoin/CHANGELOG.md:21]: Removed and is now the line posted in thread above.

**Review by sanket1729 — APPROVED** [review]
reACK ffee8ad81ba16d3ccc15b82e898adc3a67931f76

**Comment by tcharding** [comment]
reack please @sanket1729 and @Kixunil if you are still about - I rekon we can still get this out today!

**Review by Kixunil — APPROVED** [review]
ACK ffee8ad81ba16d3ccc15b82e898adc3a67931f76

**Review by apoelstra — APPROVED** [review]
ACK ffee8ad81ba16d3ccc15b82e898adc3a67931f76

**Comment by tcharding** [comment]
Wo ho ho, getting close ...

**Comment by apoelstra** [comment]
Just started merge process. should be less than 30 minutes.

**Comment by tcharding** [comment]
Can you merge the blog post PR now then? I can deploy if you like (if I do it then I can fix it if its not working :)

**Comment by tcharding** [comment]
In order for the blog post to be live when we release I'm going to deploy it now.

**Comment by apoelstra** [comment]
@tcharding yes, the remaining thing to do is merge this PR :). But I have to run my CI job on the merge commit before I can do that.

**Comment by apoelstra** [comment]
codex32 https://github.com/BlockstreamResearch/codex32/tree/master/reference/rust-codex32 has a `Fe` type which is the same as `u5` in this library.

It's not obvious that either lib should depend on the other though so actually sharing a type doesn't seem clear.

**Comment by sanket1729** [comment]
Let's go!!!

**Comment by apoelstra** [comment]
@sanket1729 sorry, we have one more thing to fix :) the `readme` key in Cargo.toml needs us to add a `../` because we moved the workspace.

Tobin will PR, then if we can quickl ACK (I won't bother doing any tests) we should be able to push in a few minutes.

**Comment by tcharding** [comment]
#1723

**Comment by apoelstra** [comment]
And we're live!

**Comment by sanket1729** [comment]
Is the website also up? I would like tweet about the release.

**Comment by tcharding** [comment]
I've borked the website, trying to fix now ...

**Comment by sanket1729** [comment]
Wohoo. Seems to be up now https://rust-bitcoin.github.io/blog/release-0.30.0/

**Comment by apoelstra** [comment]
Looks like a 404 to me

**Comment by tcharding** [comment]
yeah I have no idea why. I've hard reset to same commit as was up before and settings in https://github.com/rust-bitcoin/rust-bitcoin.github.io/settings/pages look correct

**Comment by tcharding** [comment]
Looks like its stuck in githubs github pages build system

**Comment by tcharding** [comment]
Its through their build system now but 404 still

**Comment by tcharding** [comment]
It was up the whole time, cache on my machine was returning the 404

**Comment by tcharding** [comment]
face palm, still 404 - this is working: https://rust-bitcoin.github.io/index.html

**Comment by tcharding** [comment]
We are live yo!

**Pull request merged**
